packages/wxt/src/core/utils/building/generate-wxt-dir.ts

```diff
@@ -236,20 +182,19 @@ async function getGlobalsDeclarationEntry(): Promise<WxtDirFileEntry> {
 }
 
 function getMainDeclarationEntry(references: WxtDirEntry[]): WxtDirFileEntry {
-  const text = [
-    '// Generated by wxt',
-    ...references.map((ref) => {
-      if ('module' in ref) return `/// <reference types="${ref.module}" />`;
-      if (!ref.tsReference) return;
-
+  const lines = ['// Generated by wxt'];
+  references.forEach((ref) => {
+    if ('module' in ref) {
+      return lines.push(`/// <reference types="${ref.module}" />`);
+    } else if (ref.tsReference) {
       const absolutePath = resolve(wxt.config.wxtDir, ref.path);
       const relativePath = relative(wxt.config.wxtDir, absolutePath);
-      return `/// <reference types="./${normalizePath(relativePath)}" />`;
-    }),
-  ].join('\n');
+      lines.push(`/// <reference types="./${normalizePath(relativePath)}" />`);
+    }
+  });
   return {
     path: 'wxt.d.ts',
-    text,
+    text: lines.join('\n') + '\n',
```

Comment:
There were blank lines being generated when a file didn't have a reference, this fixes that.